src/skirmishbunker/SeriesHelper.py

```diff
@@ -0,0 +1,110 @@
+import cadquery as cq
+from cadqueryhelper import series
+from math import floor
+
+class SeriesHelper:
+    def __init__(self):
+        # basic shape info
+        self.shape = None
+        self.outer_length = 0
+        self.outer_width = 0
+        self.length_offset = 0
+
+        # "component" (panel) size
+        self.comp_length = 0
+        self.comp_padding = 0
+
+        # additional translation
+        self.x_translate = 0
+        self.y_translate = 0
+        self.z_translate = 0
+
+        self.skip_list = []
+        self.keep_list = []
+
+        self.scene = None
+
+    def __validate(self):
+        if (self.shape == None):
+            raise Exception('No shape set')
+
+        if (self.outer_length <= 0 or self.outer_width <= 0):
+            raise Exception('Outer width and length need to be set')
+
+        if (self.comp_padding + self.comp_length <= 0):
+            raise Exception('Component sizes need to be set')
+
+    def get_scene(self):
+        if (self.scene == None):
+            raise Exception('You must ')
```

Comment:
lol, slightly better message than "You must " 😂.

I'm tempted to ask you to call this method "build" instead of "get_scene". That would fall more in line with the other factory classes.

Reply:
OKAYYYY... That did say something more comprehensive... God only knows what happened to it...